From 2850f37ecc5635adf6a28a54f6e74e6e56d1d80b Mon Sep 17 00:00:00 2001
From: Haibo Chen <haibo.chen@nxp.com>
Date: Mon, 9 Jul 2018 19:15:28 +0800
Subject: [PATCH 4133/5242] MLK-18816-1 drm/panel: put the RST pin back to
 high after the reset operation

commit  0f110eddba9909d070821f03e776f71733e00f6b from
https://source.codeaurora.org/external/imx/linux-imx.git

Panel and Touch share the same reset pin, keep reset pin in low will
make touch can't work. This patch make the reset pin keep in low for
over 15ms (to meet the RM requirement), and then put the reset pin
back to high (which will not impact touch).

Signed-off-by: Haibo Chen <haibo.chen@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/drm/panel/panel-raydium-rm67191.c |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/panel/panel-raydium-rm67191.c b/drivers/gpu/drm/panel/panel-raydium-rm67191.c
index 7cd1bf5..74dc68f 100644
--- a/drivers/gpu/drm/panel/panel-raydium-rm67191.c
+++ b/drivers/gpu/drm/panel/panel-raydium-rm67191.c
@@ -270,7 +270,8 @@ static int rad_panel_unprepare(struct drm_panel *panel)
 
 	if (rad->reset != NULL) {
 		gpiod_set_value(rad->reset, 0);
-		usleep_range(10000, 15000);
+		usleep_range(15000, 17000);
+		gpiod_set_value(rad->reset, 1);
 	}
 
 	rad->prepared = false;
-- 
1.7.9.5

